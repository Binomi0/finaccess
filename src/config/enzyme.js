import { mount, shallow, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

configure({ adapter: new Adapter() });

exports.mount = mount;
exports.shallow = shallow;
