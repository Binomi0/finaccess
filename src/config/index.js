const random = Math.floor(Math.random() * 1000);
const results = 3;

export const apikey = '2b338eb5f577d71ecf79c2347a55dc75';
export const filters = `&limit=${results}&offset=${random}`;
