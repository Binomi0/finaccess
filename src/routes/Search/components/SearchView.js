import React, { Component } from 'react';

import SearchHeroes from '../../../components/SearchHeroes';
import Character from '../../../components/Character';
import MarvelAPI from '../../../api';
import { CharacterContainer, CharacterTitle } from '../styles/styles';

class SearchView extends Component {
  state = {
    characters: [],
    selectedIndex: 0,
  };

  componentDidMount() {
    this.getCharacters();
  }

  handleClick = (selectedIndex) => {
    this.setState({ selectedIndex });
  };

  handleChange = ({ target: { value: character } }) => {
    this.getCharacters(character);
  };

  getCharacters = async (character) => {
    const characters = await MarvelAPI.characters.getCharacters(character);
    this.setState({ characters });
  };

  render() {
    const { characters, selectedIndex } = this.state;
    return (
      <div>
        <CharacterTitle>Buscador de Heroes de Marvel</CharacterTitle>
        <SearchHeroes handleChange={this.handleChange} />
        <CharacterContainer active={characters.length && 1}>
          {characters.map((character, idx) => (
            <Character
              key={character.id}
              character={character}
              handleClick={this.handleClick}
              selected={selectedIndex === idx}
              idx={idx}
            />
          ))}
        </CharacterContainer>
      </div>
    );
  }
}

export default SearchView;
