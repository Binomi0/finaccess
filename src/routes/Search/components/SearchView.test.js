import React from 'react';
import SearchView from './SearchView';
import MarvelApi from '../../../api';
import { mount } from '../../../config/enzyme';

describe('SearchView', () => {
  let character;

  beforeAll(async () => {
    character = await MarvelApi.characters.getCharacters('Thor');
  });

  it('renders SearchView Page', () => {
    if (character.length) {
      const MockedSearchView = mount(<SearchView />);

      expect(character).toHaveLength(3);
      expect(MockedSearchView.html()).toMatchSnapshot();
    } else {
      expect(character).toHaveLength(0);
    }
  });
});
