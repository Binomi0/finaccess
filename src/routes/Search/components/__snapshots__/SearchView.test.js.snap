// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchView renders SearchView Page 1`] = `""`;
